packages/react-core/src/hooks/useDropZone.ts

```diff
@@ -81,11 +81,54 @@ export default function useDropZone({
     setDragState(rejectedFiles.length > 0 ? 'reject' : 'accept');
   };
 
-  const onDrop = (event: React.DragEvent<HTMLDivElement>) => {
+  const onDrop = async (event: React.DragEvent<HTMLDivElement>) => {
     event.preventDefault();
     event.stopPropagation();
     setDragState('inactive');
-    const files = Array.from(event.dataTransfer.files);
+
+    const files: File[] = [];
+
+    // Add any directly dropped files
+    if (event.dataTransfer.files.length > 0) {
+      files.push(...Array.from(event.dataTransfer.files));
+    }
+
+    // Handle items for directory support
+    if (event.dataTransfer.items) {
```

Comment:
are `items` and `files` mutually exclusive? i wonder if we could include something twice